### PR TITLE
Improvements, issue fixes

### PR DESCRIPTION
**Pull request by subnut**
Closes psliwka/vim-smoothie#1
Closes psliwka/vim-smoothie#11
Closes psliwka/vim-smoothie#12
Closes psliwka/vim-smoothie#17
Closes psliwka/vim-smoothie#18
Closes psliwka/vim-smoothie#19
Closes psliwka/vim-smoothie#20
Closes psliwka/vim-smoothie#22

### CODE REVIEW

**Comment by subnut** [comment]
@psliwka May I request you to make me a collaborator of this repo?

PS. yep, It was me who mailed you

**Comment by subnut** [comment]
[My GeoGebra plot comparing the old algorithm with two newer ones, out of which one has been implemented](https://www.geogebra.org/classic/uttvkrxk)

**Comment by benwoodward** [comment]
Scrolling is noticeably smoother, nice!

**Comment by subnut** [comment]
Two requests to @psliwka 
- Don't _squash and merge_, do a simple merge
- Make me a collaborator

**Comment by subnut** [comment]
@psliwka I suggest we move this repo to the @vim-smoothie organization I have created for specifically this plugin.
I have sent an invitation to you to become an owner of the organization.

If you agree to my suggestion, then please have a look at the following article on how to transfer ownership  - 

https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/transferring-a-repository

I ask for transfer of ownership because it is better to have an organization for a plugin (similar to vim-airline) and because transfer of ownership ensures _smooth_ (pun intended) transition for the users, PRs, issues, etc. 
For e.g. 
https://github.com/bling/vim-airline automatically redirects to https://github.com/vim-airline/vim-airline
https://github.com/w0rp/ale redirects to https://github.com/dense-analysis/ale 
etc.

**Comment by psliwka** [comment]
Wow, this is an impressive amount of work, thanks! I'm starting to review this now. Given the volume of changes, please bear with me till I get through the end ;) I'll post any potential questions as I go.

File: autoload/smoothie.vim [hunk 1]
```diff
@@ -49,6 +49,14 @@ endfunction
 " already at the bottom.  Return 1 if cannot move any lower.
 function s:step_down()
   if line('.') < line('$')
+    " NOTE: the three lines of code following this comment block
+    " have been implemented as a temporary workaround for a vim issue
```
Comment (psliwka) [inline autoload/smoothie.vim:110]: How do we know it's temporary? Is there any ticket on Vim side we could track?

Comment (subnut) [reply autoload/smoothie.vim:110]: Well... I like to consider _issue_s as temporary.. :smile: 
To be honest.. I wrote "temporary" because we don't know whether it is intended behavior, but to me it doesn't seem like intended behavior.

BTW, I did open a ticket at neovim https://github.com/neovim/neovim/issues/13080

File: autoload/smoothie.vim [hunk 2]
```diff
@@ -24,6 +24,14 @@ if !exists('g:smoothie_break_on_reverse')
   let g:smoothie_break_on_reverse = 0
 endif
 
+if !exists('g:smoothie_bell_enabled')
+  ""
+  " Enable beeping when either end of the buffer has been reached, and we
+  " cannot proceed any further
```
Comment (psliwka) [inline autoload/smoothie.vim]: I think there is no need for a dedicated variable for that. Users already can control whether they want their terminals to beep by manipulating their `belloff` setting. Adding another one seems redundant to me.

Comment (psliwka) [reply autoload/smoothie.vim]: Oh, I see now that 1cefa9cb3ddddfa56e042de956c2fc1e30e1b34a partially fixes that, by configuring the default in a smart way. I'd go one step further and:
* Remove globally-accessible `g:smoothie_bell_enabled` completely to avoid abstraction leaks;
* Perform the `!(&belloff =~# 'all\|error')` check straight in the timer loop instead of at initialization, so that runtime changes to `belloff` are reflected too.

Comment (subnut) [reply autoload/smoothie.vim]: Ok. Will remove it in the next commit.
You finish your review, and I will make a commit containing all the changes you suggest. OK?

File: autoload/smoothie.vim (hunk 1 above)
Comment (psliwka) [reply autoload/smoothie.vim:110]: Awesome! Could you please link this ticket in the comment note?

File: autoload/smoothie.vim (hunk 2 above)
Comment (psliwka) [reply autoload/smoothie.vim]: SGTM 👍

File: autoload/smoothie.vim (hunk 1 above)
Comment (subnut) [reply autoload/smoothie.vim:110]: Will do.

File: autoload/smoothie.vim [hunk 3]
```diff
@@ -6,6 +6,12 @@
 " invokes motion (except smoothie#forwards, where it must be set to v:true)
 let s:ctrl_f_invoked = v:false
 
+if !exists('g:smoothie_disabled')
```
Comment (psliwka) [inline autoload/smoothie.vim:28]: Most plugins I've seen (f.ex. airblade/vim-gitgutter, Yggdroot/indentLine, unblevable/quick-scope) have a `g:<plugin_name>_enabled` switch. I think we should follow similar convention here for a more consistent user experience.

**Comment by subnut** [comment]
@psliwka After you finish your review, I shall push a commit of all the changes you suggest tomorrow. It's past midnight here :yawning_face: 

P.S. What's the time at your place right now? It's 00:10 here at my place...

**Comment by subnut** [comment]
> @psliwka I suggest we move this repo to the @vim-smoothie organization I have created for specifically this plugin.

Are we going to?

> I have sent an invitation to you to become an owner of the organization.

Did you receive it?

**Comment by psliwka** [comment]
> @psliwka After you finish your review, I shall push a commit of all the changes you suggest tomorrow. It's past midnight here 🥱

No worries, take your time :) 

> P.S. What's the time at your place right now? It's 00:10 here at my place...

I currently live in GMT+1, so 19:44 right now ;)

File: autoload/smoothie.vim [hunk 4]
```diff
@@ -1,3 +1,16 @@
+if exists('g:loaded_smoothie')
+  finish
+endif
+let g:loaded_smoothie = 1
```
Comment (psliwka) [inline autoload/smoothie.vim:4]: Why is this needed? I thought `autoload` scripts are sourced only once by Vim anyway?

File: plugin/smoothie.vim [hunk 5]
```diff
@@ -1,4 +1,4 @@
-if has('nvim') || has('patch-8.2.1280')
+if has('nvim') || has('patch-8.2.1978')
```
Comment (psliwka) [inline plugin/smoothie.vim:1]: Nit: I'd suggest to squash this with 62ef714f85e5230496c9894b70e33c7d781a3c57.

**Comment by subnut** [comment]
I have made the requested changes in a different branch, so had to open a new PR 
@psliwka Please check if your requests have been implemented in vim-smoothie#23
